numba/tests/npyufunc/test_gufunc.py

```diff
@@ -736,7 +736,9 @@ def foo(x, y, res):
 
         with self.assertRaises(ValueError) as raises:
             foo(x, y, res)
-        msg = ('Operand has a mismatch in one of its core dimensions')
+        msg = ('operand 0 has a mismatch with operand 1 one of its core '
+               'dimension(s), with bar signature (n),(n) -> () '
+               '(shape (5, 3, 2) is different from (3))')
```

Comment:
The information content is a great improvement, but I'm struggling to understand what this means in its current form. Is it possible to match the NumPy message (and maybe enhance it with shape info/function names)?

Reply:
I've changed the exception message to match NumPy. That would probably be less confusing(?)